core/state_processor.go

```diff
@@ -126,9 +126,14 @@ func applyTransaction(msg types.Message, config *params.ChainConfig, bc ChainCon
 	}
 
 	// Set the receipt logs and create the bloom filter.
-	receipt.Logs = statedb.GetLogs(tx.Hash())
+	blockhash := header.Hash()
+	logs := statedb.GetLogs(tx.Hash())
```

Comment:
Maybe, from an API-perspective, it would be nicer if you're forced to put in the blockhash in `statedb.GetLogs(tx.Hash(), blockHash)`. As it is now, the logs you get back are not "consensus", unless you do some manual post-processing later on. (is that correct?).

If so, better make it more difficult for us to shoot ourselves in our feet later.

Reply:
Yep agreed, fixed it